Stratis.Bitcoin/AsyncLoop.cs

```diff
@@ -19,51 +19,53 @@ public interface IAsyncLoop
 
     public class AsyncLoop : IAsyncLoop
     {
+        private readonly ILogger logger;
         readonly Func<CancellationToken, Task> loopAsync;
 
-        public AsyncLoop(string name, Func<CancellationToken, Task> loop)
+        public AsyncLoop(string name, ILogger logger, Func<CancellationToken, Task> loop)
         {
             Guard.NotEmpty(name, nameof(name));
             Guard.NotNull(loop, nameof(loop));
 
             this.Name = name;
+            this.logger = logger;
             this.loopAsync = loop;
         }
 
         public string Name { get; }
 
         public Task Run(TimeSpan? repeatEvery = null, TimeSpan? startAfter = null)
         {
-            return Run(CancellationToken.None, repeatEvery, startAfter);
+            return this.Run(CancellationToken.None, repeatEvery, startAfter);
         }
 
         public Task Run(CancellationToken cancellation, TimeSpan? repeatEvery = null, TimeSpan? startAfter = null)
         {
             Guard.NotNull(cancellation, nameof(cancellation));
 
-            return StartAsync(cancellation, repeatEvery ?? TimeSpan.FromMilliseconds(1000), startAfter);
+            return this.StartAsync(cancellation, repeatEvery ?? TimeSpan.FromMilliseconds(1000), startAfter);
         }
 
-        public static Task Run(string name, Func<CancellationToken, Task> loop, TimeSpan? repeatEvery = null, TimeSpan? startAfter = null)
+        public static Task Run(string name, ILogger logger, Func<CancellationToken, Task> loop, TimeSpan? repeatEvery = null, TimeSpan? startAfter = null)
```

Comment:
so here we have a logger at the class level and a logger passed as a parameter to this method.
Ah I see, that's because this static AsyncLoop.Run calls another new AsyncLoop.Run. 
Can we remove this static method? it's not used anywhere.